

syntax = "proto3";

package fixbug;

/*
    1:需要有请求
    2：请求有对应的响应，另外有错误码
    3：还需要好友列表
*/
// 错误码
message Resultcode
{
    int32 errorcode = 1;
    bytes errormsg = 2;
}

// 好友列表好友个数
message GetfriendListsRequest
{
    uint32 userid = 1;
}

// 响应
message GetfriendListsResponse
{
    Resultcode result = 1;
    repeated bytes friends = 2;
}

// 好友服务模块
service FriendServiceRpc
{
    // 需要重写的名字是GetfriendLists《------来源于protobuf的虚函数，需要重写这个虚函数
    rpc GetfriendLists(GetfriendListsRequest) returns (GetfriendListsResponse);
}